test/extended/operators/olm.go

```diff
@@ -16,37 +14,60 @@ var _ = g.Describe("[Feature:Platform] OLM should", func() {
 
 	var oc = exutil.NewCLI("olm", exutil.KubeConfigPath())
 
-	// The list of all available OLM resources
-	olmResources := []string{
-		"packagemanifests", "catalogsources", "clusterserviceversions",
-		"installplans", "operatorgroups", "subscriptions",
+	operators := "operators.coreos.com"
+	providedAPIs := []struct {
+		fromAPIService bool
+		group          string
+		version        string
+		plural         string
+	}{
+		{
+			fromAPIService: true,
+			group:          "packages." + operators,
+			version:        "v1",
+			plural:         "packagemanifests",
+		},
+		{
+			group:   operators,
+			version: "v1",
+			plural:  "operatorgroups",
+		},
+		{
+			group:   operators,
+			version: "v1alpha1",
+			plural:  "clusterserviceversions",
+		},
+		{
+			group:   operators,
+			version: "v1alpha1",
+			plural:  "catalogsources",
+		},
+		{
+			group:   operators,
+			version: "v1alpha1",
+			plural:  "installplans",
+		},
+		{
+			group:   operators,
+			version: "v1alpha1",
+			plural:  "subscriptions",
+		},
 	}
 
-	for i := range olmResources {
-		g.It("list "+olmResources[i], func() {
-			var resourceList map[string]interface{}
-			// Get resource yaml and parse
-			output, err := oc.AsAdmin().Run("get").Args(olmResources[i], "--all-namespaces", "-o", "yaml").Output()
-			o.Expect(err).NotTo(o.HaveOccurred())
-			err = yaml.Unmarshal([]byte(output), &resourceList)
-			if err != nil {
-				e2e.Logf("Unable to parse %s yaml list", olmResources[i])
+	for _, api := range providedAPIs {
+		g.It(fmt.Sprintf("be installed with %s at version %s", api.plural, api.version), func() {
+			if api.fromAPIService {
+				// Ensure spec.version matches expected
+				raw, err := oc.AsAdmin().Run("get").Args("apiservices", fmt.Sprintf("%s.%s", api.version, api.group), "-o=jsonpath='{spec.version}'").Output()
+				o.Expect(err).NotTo(o.HaveOccurred())
+				o.Expect(raw).To(o.Equal(api.version))
+			} else {
+				// Ensure expected version exists in spec.versions and is both served and stored
+				raw, err := oc.AsAdmin().Run("get").Args("crds", fmt.Sprintf("%s.%s", api.plural, api.group), fmt.Sprintf("-o=jsonpath='{spec.versions[?(@.name=\"%s\")]}'", api.version)).Output()
```

Comment:
I guess this JSON filter should be `-o=jsonpath='{.spec.versions[?(@.name==\"%s\")]}'`.